src/Momento.Sdk.Incubating/Requests/CollectionTtl.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+namespace Momento.Sdk.Incubating.Requests
+{
+    /// <summary>
+    /// Represents the desired behavior for managing the TTL on collection
+    /// objects (dictionaries, lists, sets) in your cache.
+    ///
+    /// For cache operations that modify a collection, there are a few things
+    /// to consider.  The first time the collection is created, we need to
+    /// set a TTL on it.  For subsequent operations that modify the collection
+    /// we may choose to update the TTL in order to prolong the life of the
+    /// cached collection object, or we may choose to leave the TTL unmodified
+    /// in order to ensure that the collection expires at the original TTL.
+    ///
+    /// The default behavior is to refresh the TTL (to prolong the life of the
+    /// collection) each time it is written.  This behavior can be modified
+    /// by calling the <see cref="noRefreshTtlOnUpdates"/>
+    /// </summary>
+    /// 
+    /// <param name="Ttl">The TimeSpan after which the cached collection
+    /// should be expired from the cache.  If <code>null</code>, we use the default
+    /// TTL TimeSpan that was passed to the <see cref="SimpleCacheClient"/>constructor</param>.
+    /// <param name="RefreshTtl">If true, the collection's TTL will be refreshed (to
+    /// prolong the life of the collection) on every update.  If false, the collection's
+    /// TTL will only be set when the collection is initially created.</param>
+    public record struct CollectionTtl(TimeSpan? Ttl = null, bool RefreshTtl = true)
+    {
+        /// <summary>
+        /// The default way to handle TTLs for collections.  The default TTL
+        /// TimeSpan that was specified when constructing the <see cref="SimpleCacheClient"/>
+        /// will be used, and the TTL for the collection will be refreshed any
+        /// time the collection is modified.
+        /// </summary>
+        /// <returns></returns>
+        public static CollectionTtl fromCacheTtl()
+        {
+            return new CollectionTtl(Ttl: null, RefreshTtl: true);
+        }
+        /// <summary>
+        /// Constructs a CollectionTtl with the specified TimeSpan.  The TTL
+        /// for the collection will be refreshed any time the collection is
+        /// modified.
+        /// </summary>
+        /// <param name="ttl"></param>
+        /// <returns></returns>
+        public static CollectionTtl of(TimeSpan ttl)
+        {
+            return new CollectionTtl(Ttl: ttl);
+        }
+
+        /// <summary>
+        /// Specifies that the TTL for the collection should be refreshed when
+        /// the collection is modified.  (This is the default behavior.)
+        /// </summary>
+        /// <returns></returns>
+        public CollectionTtl refreshTtlOnUpdates()
```

Comment:
would it be clearer that this isn't a mutative function if this was "withRefreshTtlOnUpdates"? I often see "with" methods follow this pattern of returning a copy of myself, just with 1 thing changed.
It's a thing people might run into, calling refreshTtlOnUpdates() and dropping the return value, thinking it was set on the original object. I don't think c# has [a convenient thing like rust has](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute) to help prevent this kind of naive misuse.

Reply:
huh, yeah that's a good point.  I was just trying to keep the names from getting any longer but I think i agree with you.

Reply:
Agreed about the naming. Either we make the names more explicit or we turn this into a formal builder: `CollectionTtlBuilder.Ttl(ttl).NoRefreshOnUpdates().build()`